src/transport/raw/TCP.cpp

```diff
@@ -336,7 +336,6 @@ CHIP_ERROR TCPBase::StartConnect(const PeerAddress & addr, Transport::AppTCPConn
 
     mUsedEndPointCount++;
 
-    endPointHolder.release();
```

Comment:
Wondering what was the reasoning behind this removal?

Reply:
This is the other side of https://github.com/project-chip/connectedhomeip/pull/41198#discussion_r2400306603 - i.e. we now decrement the counter on dtor & `ActiveTCPConnectionState` does its own ref count